packages/dom-module-styling/dom-module-styling.js

```diff
@@ -0,0 +1,63 @@
+import { DomModule } from '@polymer/polymer/lib/elements/dom-module.js';
+import { stylesFromTemplate } from '@polymer/polymer/lib/utils/style-gather.js';
+import { unsafeCSS } from 'lit';
+
+window.Vaadin = window.Vaadin || {};
+window.Vaadin.domModuleStyling = {};
+
+function getModuleStyles(module) {
+  return stylesFromTemplate(module.querySelector('template')).map((styleElement) => {
+    return unsafeCSS(styleElement.textContent);
+  });
+}
+
+let moduleIdIndex = 0;
+// Map of <CSSResult, Polymer.DomModule> pairs.
+const styleMap = {};
+
+window.Vaadin.domModuleStyling.registerStyles = (themeFor, styles = [], options) => {
+  const themeId = (options && options.moduleId) || `custom-style-module-${moduleIdIndex++}`;
+
+  const processedStyles = styles.map((cssResult) => {
+    if (!styleMap[cssResult]) {
+      const template = document.createElement('template');
```

Comment:
I'm not sure if we actually need creating the template here. The whole logic of [`stylesFromTemplate`](https://github.com/Polymer/polymer/blob/10220c9af132149c31cd0539273aad1be251fcb4/lib/utils/style-gather.js#L113) is about handling `<style include="">` and resolving URLs when `baseDir` is provided (this was a thing in Polymer 2 with HTML Imports). We don't have any of these cases with `registerStyles`.

Reply:
Cleaned up `registerStyles`. The function still creates the template and maps `options.include` to a `<style include="...">` because it's possible to include styles from a `<dom-module>` that's created without using `registerStyles`.